### PR TITLE
ci: CI Testing

### LINKED ISSUE #29: Setup unit / integration tests
We aren't currently running any tests. We should configure unit / integration tests as documented here: https://code.visualstudio.com/api/working-with-extensions/testing-extension

also should setup CI to run them via GH actions:
https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions

This looks interesting as a possibility for better integration tests:
https://github.com/stateful/vscode-awesome-ux/blob/main/docs/TestingExtensions.md

### PR DESCRIPTION
**Pull request by bmingles**
- Setup CI for unit + e2e tests
- PR shows summary comments added for 

resolves #29

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,5 +1,13 @@
 {
   "singleQuote": true,
   "trailingComma": "es5",
-  "arrowParens": "avoid"
+  "arrowParens": "avoid",
+  "overrides": [
```
Comment (bmingles) [inline .prettierrc:5]: Apparently newer versions of Prettier add trailing commas to jsonc files which is not always supported.

Comment (mofojed) [reply .prettierrc:5]: I think this applies to `.json` files as well, no? I know `package.json` is not happy if you add a trailing comma

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,45 @@
 # Contributing to Deephaven in VS Code
 
+## Development
+
+### Unit Testing
+
+TODO:
```
Comment (mofojed) [inline CONTRIBUTING.md:7]: TODO - fill in or write a ticket

**Review by mofojed — CHANGES_REQUESTED** [review]
Just a couple things to double check

File: CONTRIBUTING.md (hunk 2 above)
Comment (bmingles) [reply CONTRIBUTING.md:7]: Yes, the prettierrc thing does apply to json, but that already works as expected. The issue here is that prettier incorrectly adds trailing commas to jsonc. It doesn't do that for json

File: package.json [hunk 3]
```diff
@@ -27,8 +27,8 @@
   "main": "./out/extension.js",
   "scripts": {
     "clean": "rm -rf out",
-    "test": "npm run ts:build && npm run ts:check && npm run test:lint && npm run test:unit",
-    "test:ci": "npm run test",
+    "test": "npm run test:unit",
+    "test:ci": "npm run ts:build && npm run ts:check && npm run test:lint && npm run test:unit",
```
Comment (mofojed) [inline package.json:31]: Should consider using `concurrently`, though this works.

**Review by mofojed — APPROVED** [review]

**Pull request merged**